test_fms/axis_utils/test_axis_utils.F90

```diff
@@ -385,28 +389,36 @@ subroutine test_frac_index_fail
   ret_test = frac_index(1.5_k, values)
 end subroutine
 
-subroutine test_nearest_index
+subroutine test_nearest_index(increasing_array)
+  logical, intent(in) :: increasing_array !< .True. if test using an increasing array
   real(k) :: arr(5)
+  integer :: ans(12)
 
-  arr = [5._k, 12._k, 20._k, 40._k, 100._k]
+  if (increasing_array) then
+    arr = [5._k, 12._k, 20._k, 40._k, 100._k]
+    ans=(/1, 5, 1, 2, 3, 4, 5, 1, 2, 2, 3, 3/)
+  else
+    arr = [100._k, 40._k, 20._k, 12._k, 5._k]
+    ans=(/5, 1, 5, 4, 3, 2, 1, 5, 4, 4, 3, 3/)
+  endif
```

Comment:
Is there a designed fail test for a non-monotonic array?  I don't see one and think it might be a good one to add.

Reply:
We already have a test for that:
https://github.com/NOAA-GFDL/FMS/blob/06b94a7f574e7794684b8584391744ded68e2989/test_fms/axis_utils/test_axis_utils.F90#L427-L434

Reply:
Thanks - I didn't a separate function for that and thought it would be included in the same area is increasing vs decreasing.